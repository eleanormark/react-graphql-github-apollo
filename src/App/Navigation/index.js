import React, { useState } from "react";
import { Link, withRouter } from "react-router-dom";

import * as routes from "../../constants/routes";
import Button from "../../Button";

import './style.css'
import styled from 'styled-components';

const Input = styled.input`
  border: solid 1px #dbdbdb;
  color:black;
  text-align: left;
  padding: 10px;
  background: #f5f5f5;
  &:focus {
    outline: none;
  }
`

const Brand = styled.h1`
  margin-left: 20px;
  color: orange;
  padding-top: 10px;
`

const Navigation = ({
  location: { pathname },
  queryString,
  setQuerySearch
}) => (
  <header className="Navigation">
    <Brand>StarQuest</Brand>
    <div className="Navigation-link">
      <Link to={routes.STARRED_REPOS}>Starred Repos</Link>
    </div>
    <div className="Navigation-link">
      <Link to={routes.QUERY_TERM}>Search for Repos</Link>
    </div>

    {pathname === routes.QUERY_TERM && (
      <QuerySearch
        queryString={queryString}
        setQuerySearch={setQuerySearch}
      />
    )}
  </header>
);

const QuerySearch = (props) => {
  const [term, setTerm] = useState(props.queryString);
  return (
    <div className="Navigation-search">
      <form
        onSubmit={e => {
          props.setQuerySearch(term);
          e.preventDefault();
        }}
      >
        <Input
          type="text"
          value={term}
          onChange={e => setTerm(e.target.value)}
        />{" "}
        <Button black type="submit">
          Search
        </Button>
      </form>
    </div>
  );
};
export default withRouter(Navigation);
