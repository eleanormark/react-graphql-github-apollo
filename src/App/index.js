    
import React, { useEffect } from 'react';

function App() {
  useEffect(() => {
      console.log("hello from console")
  }, []);

  return (
    <div>
      <h1>hello</h1>
    </div>
  );
}

export default App;