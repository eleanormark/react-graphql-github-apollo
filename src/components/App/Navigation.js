import React, { useState } from "react";
import { Link, withRouter } from "react-router-dom";
import * as routes from "../../constants/routes";
import Button from "../Button";
import styled from "styled-components";
import { device } from "../../constants/device";

const Input = styled.input`
  border: solid 1px #dbdbdb;
  color: black;
  font-size: 14px;
  text-align: left;
  padding: 10px;
  background: #f5f5f5;
  &:focus {
    outline: none;
  }
`;

const Brand = styled.h1`
  margin-left: 20px;
  color: orange;
  padding-top: 10px;
  @media ${device.screenM} {
    margin-left: 0px;
  }
`;

const Nav = styled.header`
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
  background-color: #24292e;
  display: flex;
  align-items: baseline;
  @media ${device.screenM} {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding-bottom: 10px;
    text-align: center;
  }
`;

const NavLink = styled.div`
  font-size: 12px;
  letter-spacing: 3.5px;
  font-weight: 500;
  text-transform: uppercase;
  padding: 20px;
  a {
    text-decoration: none;
    color: #ffffff;
    &:hover {
      color: orange;
    }
  }
  @media ${device.screenS} {
    padding-bottom: 10px;
  }
`;

const NavSearch = styled.div`
  padding: 0 10px;
  @media ${device.screenS} {
    padding: 10px 10px;
  }
`;

export const Navigation = ({
  location: { pathname },
  queryString,
  setQuerySearch
}) => (
  <Nav>
    <Brand>StarQuest</Brand>
    <NavLink>
      <Link to={routes.STARRED_REPOS}>Starred Repos</Link>
    </NavLink>
    <NavLink>
      <Link to={routes.QUERY_TERM}>Search for Repos</Link>
    </NavLink>

    {pathname === routes.QUERY_TERM && (
      <QuerySearch queryString={queryString} setQuerySearch={setQuerySearch} />
    )}
  </Nav>
);

export const QuerySearch = props => {
  const [term, setTerm] = useState(props.queryString);
  return (
    <NavSearch>
      <form
        onSubmit={e => {
          props.setQuerySearch(term);
          e.preventDefault();
        }}
      >
        <Input
          type="text"
          value={term}
          onChange={e => setTerm(e.target.value)}
        />{" "}
        <Button black type="submit">
          Search
        </Button>
      </form>
    </NavSearch>
  );
};
export default withRouter(Navigation);
