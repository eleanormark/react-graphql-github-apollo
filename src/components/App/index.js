import React, { useState } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";

import Navigation from "./Navigation";
import QueryTerm from "../QueryTerm";
import StarredRepos from "../StarredRepos";

import * as routes from "../../constants/routes";
import styled from "styled-components";
import { device } from "../../constants/device";

const Container = styled.div`
  min-height: 100%;
  display: flex;
  flex-direction: column;
  font-size: 14px;
`;

const ContainerMain = styled.div`
  flex: 1;
`;

const ContainerContent = styled.div`
  margin-top: 84px;
  margin-left: 20px;
  @media ${device.screenM} {
    margin-top: 240px;
  }
  a {
    text-decoration: none;
    color: #0366d6;
  }
`;

function App() {
  const [queryString, setQueryString] = useState("elixer");

  return (
    <Router>
      <Container>
        <Navigation queryString={queryString} setQuerySearch={setQueryString} />
        <ContainerMain>
          <Route
            exact
            path={routes.QUERY_TERM}
            component={() => (
              <ContainerContent>
                <QueryTerm queryString={queryString} />
              </ContainerContent>
            )}
          />
          <Route
            exact
            path={routes.STARRED_REPOS}
            component={() => (
              <ContainerContent>
                <StarredRepos />
              </ContainerContent>
            )}
          />
        </ContainerMain>
      </Container>
    </Router>
  );
}

export default App;
