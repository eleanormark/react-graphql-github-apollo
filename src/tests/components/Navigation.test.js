import React from 'react';
import { shallow, mount} from 'enzyme';
import { Navigation } from '../../components/App/Navigation';
import 'jest-styled-components'

test("should render Navigation correctly without <QuerySearch />", () => {
    const wrapper = shallow(<Navigation location={{pathname:"/StarredRepos"}} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render Navigation correctly with <QuerySearch />", () => {
    const wrapper = shallow(<Navigation location={{pathname:"/"}} />);
    expect(wrapper).toMatchSnapshot();
});