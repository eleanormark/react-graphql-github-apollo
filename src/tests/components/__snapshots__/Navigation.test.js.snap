// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navigation correctly with <QuerySearch /> 1`] = `
<styled.header>
  <styled.h1>
    StarQuest
  </styled.h1>
  <styled.div>
    <Link
      to="/StarredRepos"
    >
      Starred Repos
    </Link>
  </styled.div>
  <styled.div>
    <Link
      to="/"
    >
      Search for Repos
    </Link>
  </styled.div>
  <QuerySearch />
</styled.header>
`;

exports[`should render Navigation correctly without <QuerySearch /> 1`] = `
<styled.header>
  <styled.h1>
    StarQuest
  </styled.h1>
  <styled.div>
    <Link
      to="/StarredRepos"
    >
      Starred Repos
    </Link>
  </styled.div>
  <styled.div>
    <Link
      to="/"
    >
      Search for Repos
    </Link>
  </styled.div>
</styled.header>
`;
